packages/claims/src/claims/claims.ts

```diff
@@ -63,27 +64,29 @@ export class Claims implements IClaims {
   }
 
   /**
-   * Verifies integrity of the claim, the claim is issued by the user
-   *  delegate and the authenticity of the issuer's signature
+   * Authenticates claim issuer and verifies that claim is present in subject document
    *
    * @param claimUrl {string}
    * @param hashFns {{ [alg: string]: (data: string) => string }}
    */
   async verify(
-    claimUrl: string, hashFns?: { [alg: string]: (data: string) => string },
+    claimUrl: string,
+    hashFns?: { [alg: string]: (data: string) => string },
```

Comment:
```suggestion
    claimUrl: string,
    hashFns?: { [alg: string]: (data: string) => string },
    cachedIssuerDoc?: IDIDDocument
```

Reply:
This requires:
1. `validateCredentialServiceEndpoint`
```
function validateCredentialServiceEndpoint(token: string, hashFns, cachedHolderDIDDocument?) {
   const service = (await this.document.readAttribute(
      { service: { serviceEndpoint: claimUrl } },
      claim.sub,
    )) as IServiceEndpoint;
    const { hash, hashAlg } = service;
    const createHash = { ...hashes, ...hashFns }[hashAlg as string];
    if (hash !== createHash(token)) {
      throw new Error('Claim was changed');
    }
}
```
2. `decodeCredential(claim: Claim, cachedIssuerDoc?: IDIDDocument)`
```
/**
 * @param claimUrl can be an input that store API accepts
 */
function decodeCredential(claimUrl: string): Claim {
  const token = await this.store.get(claimUrl);
    const claim = this.jwt.decode(token) as (IPublicClaim | IPrivateClaim) & {
      iss: string;
      sub: string;
    };
}
```
2. `verifyClaim(claim: Claim, cachedIssuerDoc?: IDIDDocument)`